packages/grafana-data/src/types/live.ts

```diff
@@ -0,0 +1,158 @@
+import { SelectableValue } from './select';
+import { Observable } from 'rxjs';
+
+/**
+ * The channel id is defined as:
+ *
+ *   ${scope}/${namespace}/${path}
+ *
+ * The scope drives how the namespace is used and controlled
+ */
+export enum LiveChannelScope {
+  DataSource = 'ds', // namespace = data source ID
+  Plugin = 'plugin', // namespace = plugin name (singleton works for apps too)
+  Grafana = 'grafana', // namespace = feature
+}
+
+/**
+ * @experimental
+ */
+export interface LiveChannelConfig<T = any> {
+  /**
+   * The path definition.  either static, or it may contain variables identifed with {varname}
+   */
+  path: string;
+
+  /**
+   * An optional description for the channel
+   */
+  description?: string;
+
+  /**
+   * When variables exist, this list will identify each one
+   */
+  variables?: Array<SelectableValue<string>>;
+
+  /**
+   * The channel keeps track of who else is connected to the same channel
+   */
+  hasPresense?: boolean;
+
+  /**
+   * This method will be defined if it is possible to publish in this channel.
+   * The function will return true/false if the current user can publish
+   */
+  canPublish?: () => boolean;
+
+  /** convert the stream message into a message that should be broadcast */
+  processMessage?: (msg: any) => T;
+}
+
+/**
+ * @experimental
+ */
+export interface LiveChannelStatus {
+  /**
+   * {scope}/{namespace}/{path}
+   */
+  id: string;
+
+  /**
+   * unix millies timestamp for the last status change
+   */
+  timestamp: number;
+
+  /**
+   * flag if the channel is activly connected to the channel.
+   * This may be false while the connections get established or if the network is lost
+   * As long as the `shutdown` flag is not set, the connection will try to reestablish
+   */
+  connected: boolean;
+
+  /**
+   * Indicate that the channel will not recived any updates and should be discarded
+   */
+  shutdown?: any;
```

Comment:
Would it be simpler to use an enum here instead? enum ConnectionState: { NotConnected, Connected, Shutdown/Closed, Error }